k-krasilnikova/SUL
# Feature/v1-199 | Implemented pending assessments page.

# Changes

### Front-end:
1. New page `Assessments` that can be reached by route `/pending-assessments` only by users with `manager` role. Page displays manager's employees courses that are in status `assessment`. Manager can both approve and decline assessments. Requests contain: user common info, title and image of the course, technologies that will be achieved after approving assessment(only techs avatar are displayed, to see skill name and score just hover on skill you want to display) and time elapsed from the moment user started the test.
2. Updated approve/decline pending course requests API. Now front-end requests work correctly and you can approve pending course with `assessment` flag.
3. Implemented API request for fetching pending assessments.
4. Implemented API request for approving/declining assessment requests.

### Backend:
1. Implemented new API endpoint: `[GET] /api/clientCourses/assessments` for users with `Manager` role. Endpoint responses with the list of pending assessments. Example: 
```
[
    {
        "_id": "626d1dc0a3b43e649ca3582d",
        "course": {
            "title": "Vue JS",
            "technologies": [
                {
                    "skill": {
                        "name": "VueJS",
                        "image": "https://pbs.twimg.com/profile_images/920561100604825600/k8sQjqio.jpg",
                        "maxScore": 5
                    },
                    "points": 3
                },
                {
                    "skill": {
                        "name": "JavaScript",
                        "image": "https://cdn-icons-png.flaticon.com/512/5968/5968292.png",
                        "maxScore": 5
                    },
                    "points": 1
                }
            ],
            "avatar": "https://hsto.org/getpro/habr/post_images/d4c/9e7/fe6/d4c9e7fe6d3eda87c2ae0c8275cf62b8.png"
        },
        "user": {
            "firstName": "Katsiaryna",
            "lastName": "Krasilnikova",
            "position": "Team Manager / Software Engineer",
            "avatar": "https://ucarecdn.com/d1542137-f9f4-496d-ba18-28525c9817d8/empl.png"
        },
        "elapsed": {
            "days": 1,
            "hours": 20,
            "minutes": 40
        }
    },
    {
        "_id": "626d3e10a3b43e649ca35e07",
        "course": {
            "title": "TESTS DANIEL",
            "technologies": [
                {
                    "skill": {
                        "name": "English",
                        "image": "https://cdn-icons-png.flaticon.com/512/197/197374.png",
                        "maxScore": 6
                    },
                    "points": 1
                },
                {
                    "skill": {
                        "name": "Java",
                        "image": "https://cdn-icons-png.flaticon.com/512/5968/5968282.png",
                        "maxScore": 5
                    },
                    "points": 2
                },
                {
                    "skill": {
                        "name": "TypeScript",
                        "image": "https://cdn-icons-png.flaticon.com/512/5968/5968381.png",
                        "maxScore": 5
                    },
                    "points": 1
                },
                {
                    "skill": {
                        "name": "JavaScript",
                        "image": "https://cdn-icons-png.flaticon.com/512/5968/5968292.png",
                        "maxScore": 5
                    },
                    "points": 3
                },
                {
                    "skill": {
                        "name": "ReactJS",
                        "image": "https://cdn-icons-png.flaticon.com/512/45/45082.png",
                        "maxScore": 5
                    },
                    "points": 2
                },
                {
                    "skill": {
                        "name": "CSS",
                        "image": "https://cdn-icons-png.flaticon.com/512/174/174854.png",
                        "maxScore": 5
                    },
                    "points": 1
                },
                {
                    "skill": {
                        "name": "HTML",
                        "image": "https://cdn-icons-png.flaticon.com/512/569/569835.png",
                        "maxScore": 5
                    },
                    "points": 3
                },
                {
                    "skill": {
                        "name": "PHP",
                        "image": "https://cdn-icons-png.flaticon.com/512/5968/5968332.png",
                        "maxScore": 5
                    },
                    "points": 2
                },
                {
                    "skill": {
                        "name": "Python",
                        "image": "https://cdn-icons-png.flaticon.com/512/5968/5968350.png",
                        "maxScore": 7
                    },
                    "points": 1
                },
                {
                    "skill": {
                        "name": "Scala",
                        "image": "https://cdn-icons-png.flaticon.com/512/919/919834.png",
                        "maxScore": 6
                    },
                    "points": 2
                },
                {
                    "skill": {
                        "name": "MySQL",
                        "image": "https://cdn-icons-png.flaticon.com/512/1199/1199128.png",
                        "maxScore": 3
                    },
                    "points": 1
                }
            ],
            "avatar": "https://pbs.twimg.com/media/EAyWcddWkAAD_tO.jpg"
        },
        "user": {
            "firstName": "Katsiaryna",
            "lastName": "Krasilnikova",
            "position": "Team Manager / Software Engineer",
            "avatar": "https://ucarecdn.com/d1542137-f9f4-496d-ba18-28525c9817d8/empl.png"
        },
        "elapsed": {
            "days": 1,
            "hours": 18,
            "minutes": 25
        }
    }
]
```

# How to test

1. Log in with `manager` credentials.
2. Navigate to pending assessment page by clicking menu link.
3. Check the displaying of requests.
4. Hover on any skill you want to see info about.
5. Click on `Approve` button to approve assessment.
6. Click on `Decline` button to decline assessment.
